src/services/lazy-dep.ts

```diff
@@ -21,17 +21,17 @@ export async function lazyLoader(promiseGenerators: (() => Promise<any>)[]) {
   const failures = [];
   do {
     logger.info('Starting a loop...');
-    const results = await Promise.allSettled(generatorsToRun.map(g => g()));
     const generatorsToRerun = [];
-    for (let i = 0; i < results.length; i++) {
-      if (results[i].status === 'fulfilled') {
-        continue;
+    for (const g of generatorsToRun) {
+      try {
+        await g();
+      } catch (e: any) {
+        logger.info(`error object ${JSON.stringify(e)}`)
+        const err = new Error((e ?? 'An unexpected error occurred'));
+        err.stack = e.stack ?? err.stack;
```

Comment:
Nit. I think this could be:
```suggestion
        const err = e ?? new Error('An unexpected error occurred');
```

Reply:
Though I see the weird structure was there before. I wonder where it came from and why.